### ISSUE TITLE
ArrayOutOfBoundsException when using Field.getName

### ISSUE THREAD

**Issue by kaeawc** [mixpanel/mixpanel-android#241]
During initialization the SDK users some reflection [here](https://github.com/mixpanel/mixpanel-android/blob/master/src/main/java/com/mixpanel/android/mpmetrics/ResourceReader.java#L89)

I've observed a couple occurrences where this can result in an ArrayOutOfBoundsException. I have very little insight or understanding of the internals of Art, but it seems like it could be safe to catch that exception as well.

Here's the stacktrace from an app using Mixpanel SDK 4.6.1

```
java.lang.RuntimeException: Unable to create application <redacted>: java.lang.ArrayIndexOutOfBoundsException: length=15; index=56585
at android.app.ActivityThread.handleBindApplication(ActivityThread.java:4720)
at android.app.ActivityThread.access$1600(ActivityThread.java:163)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1419)
at android.os.Handler.dispatchMessage(Handler.java:102)
at android.os.Looper.loop(Looper.java:135)
at android.app.ActivityThread.main(ActivityThread.java:5536)
at java.lang.reflect.Method.invoke(Native Method)
at java.lang.reflect.Method.invoke(Method.java:372)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1397)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1192)
Caused by: java.lang.ArrayIndexOutOfBoundsException: length=15; index=56585
at java.lang.reflect.ArtField.getNameNative(Native Method)
at java.lang.reflect.ArtField.getName(ArtField.java:91)
at java.lang.reflect.Field.getName(Field.java:122)
at com.mixpanel.android.mpmetrics.ResourceReader.readClassIds(ResourceReader.java:89)
at com.mixpanel.android.mpmetrics.ResourceReader.initialize(ResourceReader.java:120)
at com.mixpanel.android.mpmetrics.ResourceReader$Ids.<init>(ResourceReader.java:23)
at com.mixpanel.android.viewcrawler.ViewCrawler$ViewCrawlerHandler.<init>(ViewCrawler.java:283)
at com.mixpanel.android.viewcrawler.ViewCrawler.<init>(ViewCrawler.java:78)
at com.mixpanel.android.mpmetrics.MixpanelAPI.constructUpdatesFromMixpanel(MixpanelAPI.java:1330)
at com.mixpanel.android.mpmetrics.MixpanelAPI.<init>(MixpanelAPI.java:218)
at com.mixpanel.android.mpmetrics.MixpanelAPI.getInstance(MixpanelAPI.java:288)
```

**Comment by sandy-8925** [comment]
Some of our users also face the same issue, with the same stacktrace as above (Mixpanel version 4.5.3)

java.lang.ArrayIndexOutOfBoundsException: length=0; index=67894
1	at java.lang.reflect.ArtField.getNameNative(Native Method)
2	at java.lang.reflect.ArtField.getName(ArtField.java:91)
3	at java.lang.reflect.Field.getName(Field.java:122)
4	at com.mixpanel.android.mpmetrics.ResourceReader.readClassIds(ResourceReader.java:85)
5	at com.mixpanel.android.mpmetrics.ResourceReader.initialize(ResourceReader.java:116)
6	at com.mixpanel.android.mpmetrics.ResourceReader$Ids.&lt;init&gt;(ResourceReader.java:19)
7	at com.mixpanel.android.viewcrawler.ViewCrawler.&lt;init&gt;(ViewCrawler.java:66)
8	at com.mixpanel.android.mpmetrics.MixpanelAPI.constructUpdatesFromMixpanel(MixpanelAPI.java:1094)
9	at com.mixpanel.android.mpmetrics.MixpanelAPI.&lt;init&gt;(MixpanelAPI.java:147)
10	at com.mixpanel.android.mpmetrics.MixpanelAPI.getInstance(MixpanelAPI.java:211)

**Comment by fcduarte** [comment]
Here's another one, from 4.6.4:

````
Caused By: java.lang.ArrayIndexOutOfBoundsException·length=14; index=31289
ArtField.java:-2 java.lang.reflect.ArtField.getNameNative	
ArtField.java:91 java.lang.reflect.ArtField.getName	
Field.java:122 java.lang.reflect.Field.getName	
SourceFile:89 com.mixpanel.android.mpmetrics.ResourceReader.readClassIds	
SourceFile:120 com.mixpanel.android.mpmetrics.ResourceReader.initialize	
SourceFile:23 com.mixpanel.android.mpmetrics.ResourceReader$Ids.<init>	
SourceFile:264 com.mixpanel.android.viewcrawler.ViewCrawler$ViewCrawlerHandler.<init>	
SourceFile:74 com.mixpanel.android.viewcrawler.ViewCrawler.<init>	
SourceFile:1331 com.mixpanel.android.mpmetrics.MixpanelAPI.constructUpdatesFromMixpanel	
SourceFile:219 com.mixpanel.android.mpmetrics.MixpanelAPI.<init>	
SourceFile:289 com.mixpanel.android.mpmetrics.MixpanelAPI.getInstance
````

If it's worth to mention it happened on a 5.1.1 device, Samsung SM-G360T

The bug has been here for a year, any plans to fix it?

**Comment by patedit** [comment]
Hi folks, this is a system crash and not caused by the library, so we can't recover from it. I've added a try/catch so the app doesn't crash in the future

**Comment by patedit** [comment]
Available on v5.6.7
